jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -529,8 +529,59 @@ public StringBuffer append (float value) {
  * @param		value	the integer
  * @return		this StringBuffer
  */
-public StringBuffer append (int value) {
+public synchronized StringBuffer append(int value) {
+	/*[IF Sidecar19-SE]*/
+	/*[IF Sidecar19-SE-B174]*/
+	int currentLength = lengthInternalUnsynchronized();
+	int currentCapacity = capacityInternal();
+
+	int valueLength = Integer.stringSize(value);
+
+	int newLength = currentLength + valueLength;
+
+	if (newLength > currentCapacity) {
+		ensureCapacityImpl(newLength);
+	}
+
+	if (String.enableCompression && count >= 0) {
+		Integer.getChars(value, newLength, this.value);
+	} else {
+		StringUTF16.getChars(value, newLength, this.value);
+	}
+
+	count = newLength;
+
+	return this;
+	/*[ELSE]*/
 	return append(Integer.toString(value));
+	/*[ENDIF]*/
+	/*[ELSE]*/
+	if (value != Integer.MIN_VALUE) {
+		if (String.enableCompression && count >= 0) {
+			return append(Integer.toString(value));
+		} else {
+			int currentLength = lengthInternalUnsynchronized();
+			int currentCapacity = capacityInternal();
+
+			int valueLength = value < 0 ? Integer.stringSize(-value) + 1 : Integer.stringSize(value);
```

Comment:
Simplify?

Reply:
@keithc-ca not sure this is possible. I left a comment on the previous review at https://github.com/eclipse/openj9/pull/713#discussion_r155064095. Could you elaborate what exactly should be simplified here?

Reply:
```
int valueLength;
if (value < 0) {
  /* stringSize can't handle negative numbers in Java 8 */
  valueLength = Integer.stringSize(-value) + 1;
} else {
  valueLength = Integer.stringSize(value);
}
```
Has the added benefit of being easier to debug as it becomes possible to set a breakpoint on a single side of the if rather than only on the whole expression.